addon/mixin.js

```diff
@@ -71,7 +72,6 @@ export default Ember.Mixin.create({
       return;
     }
 
-    this.propertyWillChange(key);
```

Comment:
I did have a quick question.  Is there a reason we are setting `propertyWillChange(key)` here and then doing some work before calling `propertyDidChange(key);`.

Asking because `notifyPropertyChange(this, key)` will call these two functions internally, but the work happens in a different order with this PR.

Reply:
I don't recall, TBH. I can't think of any practical reason that this change will be an issue. Back when this was written, I think there were various willChange and didChange hooks in Ember from Sproutcore, and you might use a willChange hook to record an old value, for example. Since that's not really a thing anymore, I think this will be fine.